tests/System.Buffers.ReaderWriter.Tests/ReaderBasicTests.cs

```diff
@@ -395,6 +395,70 @@ public void ReaderIndexIsCorrect()
             Assert.Equal(buffer.Length, reader.Consumed);
         }
 
+        [Theory,
+            InlineData(1),
+            InlineData(2),
+            InlineData(3)]
+        public void Advance_PositionIsCorrect(int advanceBy)
+        {
+            // Check that advancing through the reader gives the same position
+            // as returned directly from the buffer.
+
+            ReadOnlySequence<T> buffer = Factory.CreateWithContent(GetInputData(10));
+            BufferReader<T> reader = new BufferReader<T>(buffer);
+
+            SequencePosition readerPosition = reader.Position;
+            SequencePosition bufferPosition = buffer.GetPosition(0);
+            Assert.Equal(readerPosition.GetInteger(), bufferPosition.GetInteger());
+            Assert.Same(readerPosition.GetObject(), readerPosition.GetObject());
+
+            for (int i = advanceBy; i <= buffer.Length; i += advanceBy)
+            {
+                reader.Advance(advanceBy);
+                readerPosition = reader.Position;
+                bufferPosition = buffer.GetPosition(i);
+                Assert.Equal(readerPosition.GetInteger(), bufferPosition.GetInteger());
+                Assert.Same(readerPosition.GetObject(), readerPosition.GetObject());
+            }
+        }
+
+        [Fact]
+        public void AdvanceTo()
+        {
+            // Ensure we can advance to each of the items in the buffer
+
+            T[] inputData = GetInputData(10);
+            ReadOnlySequence<T> buffer = Factory.CreateWithContent(inputData);
+
+            for (int i = 0; i < buffer.Length; i++)
+            {
+                BufferReader<T> reader = new BufferReader<T>(buffer);
+                Assert.True(reader.TryAdvanceTo(inputData[i], advancePastDelimiter: false));
+                Assert.True(reader.TryPeek(out T value));
+                Assert.Equal(inputData[i], value);
+            }
+        }
+
+        [Fact]
+        public void AdvanceTo_AdvancePast()
```

Comment:
Can we merge AdvanceTo and AdvanceTo_AdvancePast tests and pass in the advancePastDelimiter bool? The tests are very similar.

Reply:
I considered it, but it would have been more complicated to put the conditions in than was worth it in my opinion.

Reply:
You do it in other tests like TryReadTo_Sequence (below).